src/main/java/org/sunbird/searchby/service/SearchByService.java

```diff
@@ -42,10 +44,49 @@
 			logger.info("Initializing/Refreshing the Cache Value.");
 			updateCompetencyDetails(authUserToken);
 		}
-
+		updateCompetencyDetails(authUserToken);
 		return (Collection<CompetencyInfo>) cacheManager.getCache(COMETENTY_CACHE_NAME);
 	}
 
+	public Collection<CompetencyInfo> getCompetencyDetailsByFilter(String authUserToken, SearchByFilter filter) throws Exception {
+		Object object = cacheManager.getCache(COMETENTY_CACHE_NAME);
+		if (object == null) {
+			logger.info("Initializing/Refreshing the Cache Value.");
+			updateCompetencyDetails(authUserToken);
+		}
+		Collection<CompetencyInfo> beforeFilter=(Collection<CompetencyInfo>) cacheManager.getCache(COMETENTY_CACHE_NAME);
+		Collection<CompetencyInfo> afterFilter = new ArrayList<>();
+		if(filter.getCompetencyName().isEmpty()&& filter.getCompetencyType().isEmpty() && filter.getCompetencyArea().isEmpty()){
+			return beforeFilter;
+		}else{
+			for(CompetencyInfo eachInfo: beforeFilter){
+				if(!filter.getCompetencyName().isEmpty()){
+					if (this.listToLowerCase(filter.getCompetencyName()).contains(this.stringToLowerCase(eachInfo.getName()))){
+						afterFilter.add(eachInfo);
+					}
+				}
+				if(!filter.getCompetencyType().isEmpty()){
+					if ((this.listToLowerCase(filter.getCompetencyType()).contains(this.stringToLowerCase(eachInfo.getCompetencyType())))){
+						afterFilter.add(eachInfo);
+					}
+				}
+				if(!filter.getCompetencyArea().isEmpty()){
+					if ((this.listToLowerCase(filter.getCompetencyArea()).contains(this.stringToLowerCase(eachInfo.getCompetencyArea())))){
+						afterFilter.add(eachInfo);
+					}
+				}
+			}
+		}
+		return afterFilter;
+	}
```

Comment:
Looks like there is a possibility of having duplicate objects.
Filter contains - name:nameOne, type:typeOne, area:areaOne
Imagine there are 15 objects with 'nameOne'... those gets added into afterFilter. 
if there are 30 objects with "typeOne" -- which also has few of "nameOne" objects - both gets added into the list ??